ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTextViewManager.java

```diff
@@ -80,6 +81,22 @@ public void setLineHeight(ReactTextView view, float lineHeight) {
     }
   }
 
+  @ReactProp(name = ViewProps.TEXT_DECORATION_LINE)
+  public void setTextDecorationLine(ReactTextView view, @Nullable String textDecorationLine) {
+    int paintFlags = view.getPaintFlags();
+    if (textDecorationLine == null || "none".equals(textDecorationLine)) {
+      view.setPaintFlags(paintFlags & ~Paint.UNDERLINE_TEXT_FLAG & ~Paint.STRIKE_THRU_TEXT_FLAG);
+    } else if ("underline".equals(textDecorationLine)) {
+      view.setPaintFlags(paintFlags | Paint.UNDERLINE_TEXT_FLAG & ~Paint.STRIKE_THRU_TEXT_FLAG);
+    } else if ("line-through".equals(textDecorationLine)) {
+      view.setPaintFlags(paintFlags & ~Paint.UNDERLINE_TEXT_FLAG | Paint.STRIKE_THRU_TEXT_FLAG);
+    } else if ("underline line-through".equals(textDecorationLine)) {
```

Comment:
I did consider that, but I weighed precise compatibility with the iOS side as more valuable than the future flexibility.  It prevents an Android developer from writing a style (eg. 'line-through underline') that works only on Android without being aware of it.

That said, I'd be happy to add compatible flexible parsing to both platforms.  Do you think it would make sense to include that change in this PR?